Redefinition errors for cpuid functions

Since commit 940f32c1a892dd194c3c6d66bb3550149de919fc when compilling the Tracy library on my system (Linux, KDE Neon - based on Ubuntu 20.04) I get the following compile error (issuing `make` from `library/unix`):
```
In file included from ../../public/client/TracySysTrace.cpp:612,
                 from ../../public/TracyClient.cpp:26:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:223:1: error: redefinition of ‘unsigned int __get_cpuid_max(unsigned int, unsigned int*)’
  223 | __get_cpuid_max (unsigned int __ext, unsigned int *__sig)
      | ^~~~~~~~~~~~~~~
In file included from ../../public/client/TracyProfiler.cpp:109,
                 from ../../public/TracyClient.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:223:1: note: ‘unsigned int __get_cpuid_max(unsigned int, unsigned int*)’ previously defined here
  223 | __get_cpuid_max (unsigned int __ext, unsigned int *__sig)
      | ^~~~~~~~~~~~~~~
In file included from ../../public/client/TracySysTrace.cpp:612,
                 from ../../public/TracyClient.cpp:26:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:278:1: error: redefinition of ‘int __get_cpuid(unsigned int, unsigned int*, unsigned int*, unsigned int*, unsigned int*)’
  278 | __get_cpuid (unsigned int __leaf,
      | ^~~~~~~~~~~
In file included from ../../public/client/TracyProfiler.cpp:109,
                 from ../../public/TracyClient.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:278:1: note: ‘int __get_cpuid(unsigned int, unsigned int*, unsigned int*, unsigned int*, unsigned int*)’ previously defined here
  278 | __get_cpuid (unsigned int __leaf,
      | ^~~~~~~~~~~
In file included from ../../public/client/TracySysTrace.cpp:612,
                 from ../../public/TracyClient.cpp:26:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:295:1: error: redefinition of ‘int __get_cpuid_count(unsigned int, unsigned int, unsigned int*, unsigned int*, unsigned int*, unsigned int*)’
  295 | __get_cpuid_count (unsigned int __leaf, unsigned int __subleaf,
      | ^~~~~~~~~~~~~~~~~
In file included from ../../public/client/TracyProfiler.cpp:109,
                 from ../../public/TracyClient.cpp:23:
/usr/lib/gcc/x86_64-linux-gnu/9/include/cpuid.h:295:1: note: ‘int __get_cpuid_count(unsigned int, unsigned int, unsigned int*, unsigned int*, unsigned int*, unsigned int*)’ previously defined here
  295 | __get_cpuid_count (unsigned int __leaf, unsigned int __subleaf,
      | ^~~~~~~~~~~~~~~~~

```

--- a/public/client/TracyCpuid.hpp
+++ b/public/client/TracyCpuid.hpp
@@ -1,0 +1,12 @@
+#ifndef __TRACYCPUID_HPP__
+#define __TRACYCPUID_HPP__
+
+// Prior to GCC 11 the cpuid.h header did not have any include guards and thus
+// including it more than once would cause a compiler error due to symbol
+// redefinitions. In order to support older GCC versions, we have to wrap this
+// include between custom include guards to prevent this issue.
+// See also https://github.com/wolfpld/tracy/issues/452
+
+#include <cpuid.h>
+
+#endif

--- a/public/client/TracyProfiler.cpp
+++ b/public/client/TracyProfiler.cpp
@@ -106,7 +106,7 @@ extern "C" typedef BOOL (WINAPI *t_GetLogicalProcessorInformationEx)( LOGICAL_PR
 #endif
 
 #if !defined _WIN32 && ( defined __i386 || defined _M_IX86 || defined __x86_64__ || defined _M_X64 )
-#  include <cpuid.h>
+#  include "TracyCpuid.hpp"
 #endif
 
 #if !( ( defined _WIN32 && _WIN32_WINNT >= _WIN32_WINNT_VISTA ) || defined __linux__ )

--- a/public/client/TracySysTrace.cpp
+++ b/public/client/TracySysTrace.cpp
@@ -609,7 +609,7 @@ void SysTraceGetExternalName( uint64_t thread, const char*& threadName, const ch
 #    include <sys/syscall.h>
 
 #    if defined __i386 || defined __x86_64__
-#      include <cpuid.h>
+#      include "TracyCpuid.hpp"
 #    endif
 
 #    include "TracyProfiler.hpp"

